fix: guard setBasicFunctionality calls with state 2 feature flag

fabiobozzo opened the pull request:
## **Description**

Original Slack thread: https://consensys.slack.com/archives/C08R8HGNFDH/p1758038233910489

Prevents unwanted multichain account alignment from running when State 2 (BIP-44 multichain accounts) is disabled. This fixes a release blocker where Solana accounts were being created automatically during app startup even when the feature is not enabled: 

**Root Cause Analysis:**
The issue was caused by unguarded `setBasicFunctionality` calls during app initialization that triggered `MultichainAccountService.alignWallets()` regardless of the State 2 feature flag status. This alignment process created Solana accounts to match existing EVM accounts, and the backup & sync system captured these temporary accounts before they could be properly cleaned up.

**Technical Details:**
1. App startup called `setBasicFunctionality` without feature flag guard in store initialization
2. This triggered `MultichainAccountService.alignWallets()` even when State 2 was disabled
3. Alignment created Solana accounts to match existing EVM accounts
4. Backup & sync captured these temporary accounts before cleanup

**Changes:**
- Add `isMultichainAccountsState2Enabled()` guard in store initialization (`app/store/index.ts`)
- Add same guard in `toggleBasicFunctionality` action creator (`app/actions/settings/index.js`)
- Update test mocks to handle new feature flag dependency

## **Changelog**

CHANGELOG entry: Fixed unwanted Solana Snap accounts appearing when BIP-44 multichain accounts feature is disabled

## **Related issues**

Fixes: Release blocker - unwanted Solana Snap accounts appearing in State 1 builds

## **Manual testing steps**

```gherkin
Feature: Prevent unwanted Solana account creation

  Scenario: User starts app with State 2 disabled
    Given State 2 (BIP-44 multichain accounts) feature flag is disabled
    And user has existing EVM accounts
    
    When user starts the MetaMask Mobile app
    Then no unwanted Solana Snap accounts should be created
    And only existing EVM accounts should be visible
    
  Scenario: User toggles basic functionality with State 2 disabled
    Given State 2 feature flag is disabled
    And user is in Settings
    
    When user toggles basic functionality on/off
    Then MultichainAccountService.setBasicFunctionality should not be called
    And no account alignment should occur
```

## **Screenshots/Recordings**

### **Before**
- Unwanted Solana Snap accounts appeared automatically during app startup
- Issue reproduced on main branch with State 1 configuration

### **After**
- No unwanted Solana accounts created when State 2 is disabled
- setBasicFunctionality calls are properly guarded by feature flag

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

Commit:
fix: guard setBasicFunctionality calls with State 2 feature flag

Prevents unwanted multichain account alignment from running when
State 2 (BIP-44 multichain accounts) is disabled. This fixes the
issue where Solana accounts were being created automatically
during app startup even when the feature is not enabled.

The issue was caused by:
1. App startup calling setBasicFunctionality without feature flag guard
2. This triggered MultichainAccountService.alignWallets()
3. Alignment created Solana accounts to match existing EVM accounts
4. Backup & sync captured these temporary accounts before cleanup

Changes:
- Add isMultichainAccountsState2Enabled() guard in store initialization
- Add same guard in toggleBasicFunctionality action creator
- Update test mocks to handle new feature flag dependency

Fixes release blocker where users saw unwanted Solana Snap accounts.

Commit:
fix: prettier formatting for settings/index.js

Commit:
fix: revert store initialization guard - only guard MultichainAccountService calls

The store initialization uses plain setBasicFunctionality action creator which
only updates Redux state and doesn't call MultichainAccountService.
The guard should only be in the thunk action creator to prevent the service call.

As per Charly's feedback: we should only prevent calling MultichainAccountService
when State 2 is OFF, but allow the original Redux behavior to go through.

ccharly reviewed (dismissed):
LGTM code-wise (haven't tested), also, left a nit comment about the `RemoteFeatureFlagController` state, but it's not critical.

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

Commit:
test: remove unnecessary RemoteFeatureFlagController mock

The RemoteFeatureFlagController mock is not needed since we're already
mocking the feature flag module directly and the store initialization
no longer uses the feature flag check.

As per Charly's feedback in PR review.

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

Commit:
test: add RemoteFeatureFlagController mock for feature flag tests

The test 'disables backup and sync when basic functionality is disabled'
calls toggleBasicFunctionality which requires both the feature flag module
mock and the Engine RemoteFeatureFlagController mock since the require()
call happens at runtime.

Both mocks are needed for complete test coverage.

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

Commit:
test: remove redundant RemoteFeatureFlagController mock

As per feedback from @ccharly, @gantunesr, and cursor bot:
The Engine.context.RemoteFeatureFlagController mock is redundant since
we're already mocking the feature flag module directly. The redundant
mock could hide potential issues and uses incorrect flag structure.

Keeping only the necessary feature flag module mock.

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

Commit:
test: add comprehensive coverage for toggleBasicFunctionality feature flag logic

Added dedicated test file to verify the conditional MultichainAccountService calls:

1. When State 2 is disabled: Redux state updates but service is not called
2. When State 2 is enabled: Both Redux state and service are updated
3. Error handling: Service errors are caught and logged gracefully

This provides 100% coverage of the new conditional branch logic
that prevents unwanted Solana account creation when State 2 is off.

Commit:
fix: prettier formatting for test files

Apply prettier formatting to:
- app/actions/settings/toggleBasicFunctionality.test.js
- app/components/UI/Identity/BackupAndSyncToggle/BackupAndSyncToggle.test.tsx

All tests still passing after formatting.

fabiobozzo committed:
Merge branch 'main' into fix/guard-basic-functionality-alignment

ccharly reviewed (approved):
LGTM (not tested)
